### PR TITLE
fix: update dependencies, add empty cargo workspace, comment an doc e…

### PR DESCRIPTION
**Pull request by apollo-pablo**
Fix some issues that breaks `cargo test` on other projects that use this as dependency.

### CODE REVIEW

**Comment by apollo-sturdy** [comment]
As per issue https://github.com/apollodao/cw-token/issues/13 we want to use version "0.16" for `cw-plus` dependencies and "1.1" for cosmwasm packages.

File: Cargo.toml [hunk 1]
```diff
@@ -6,16 +6,17 @@ authors = ["Apollo Devs"]
 edition = "2021"
 repository = "https://github.com/apollodao/cw-token"
 
+[workspace]
```
Comment (apollo-sturdy) [inline Cargo.toml:9]: Why is this needed?

**Review by apollo-sturdy — CHANGES_REQUESTED** [review]

File: src/traits.rs [hunk 2]
```diff
@@ -31,16 +31,17 @@ pub trait Instantiate {
...
-    /// pub fn instantiate(
-    ///     deps: DepsMut,
-    ///     env: Env,
-    ///     info: MessageInfo,
-    ///     msg: InstantiateMsg,
-    /// ) -> Result<Response, ContractError> {
-    ///     let my_token = MyToken::new(...);
-    ///     my_token.instantiate(deps, to_binary(&msg.init_info)?)
-    /// }
+    /// // TODO: fix this example
```
Comment (apollo-sturdy) [inline src/traits.rs:34]: Just use the `ignore` attribute, since this example is not meant to compile. See https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#attributes

**Review by apollo-sturdy — CHANGES_REQUESTED** [review]

File: src/traits.rs (hunk 2 above)
Comment (apollo-pacman) [reply src/traits.rs]: fixed

**Comment by apollo-pacman** [comment]
> As per issue #13 we want to use version "0.16" for `cw-plus` dependencies and "1.1" for cosmwasm packages.

I fixed this in the latest commit

File: src/traits.rs [hunk 3]
```diff
@@ -29,16 +29,16 @@ pub trait Instantiate {
     /// ## Returns
     /// Returns a Response containing the messages to instantiate the token.
     ///
-    /// ## Example
-    /// ```
+    /// ## Example (pseudocode)
+    /// ```ignore
```
Comment (apollo-pacman) [inline src/traits.rs:33]: @apollo-pablo can you verify that this works with the test framwork before we merge?

File: Cargo.toml (hunk 1 above)
Comment (apollo-pablo) [reply Cargo.toml:9]: This is need it for crate optimizoor to compile and generate wasm files, since the testing framework clone from github the branch and uses as input to the compiler, but the integration test works as a workspace and rust complains that the cloned repo doesnt belongs to him, except if you **explicit** put an empty workspace.

File: src/traits.rs (hunk 3 above)
Comment (apollo-pablo) [reply src/traits.rs:33]: I cannot verify it since this dependency is not used by the testing framework directly, but used by the contracts. But running `cargo test` on the branch work, so it is expected that this fix the problem.

**Comment by apollo-pablo** [comment]
I notice that we are using 0.16 here, keep in mind that 0.16 have some breaking changes that probably we have to apply on other contracts.

**Comment by apollo-pacman** [comment]
we have been using 0.16 in the other contracts for osmosis vaults. Gabe from mars asked us to update to 0.16 in the vault standard repo since they use 0.16 in the Mars code. Does 0.16 work for you @apollo-pablo?

File: src/traits.rs (hunk 3 above)
Comment (apollo-pacman) [reply src/traits.rs:33]: perfect! sounds good
